.gitlab-ci.yml

```diff
@@ -0,0 +1,61 @@
+variables:
+  CI_DOCKER_IMAGE: registry.ddbuild.io/images/docker@sha256:23643e6da71fc2551046a1928c071e6db5770f050a0f5a1e613168d2c65fb8bc # Docker 20.10.13
+  DOCKER_CONTEXT: "."
+  DOCKER_BUILD_ARGS: ""
+
+.build-docker-image: &build-docker-image
+  stage: build
+  image: $CI_DOCKER_IMAGE
+  tags: ["runner:docker"]
+  rules:
+    # For all branches ending with -dd, automatically push to prod
+    - if: $CI_COMMIT_BRANCH =~ /.*-dd/
```

Comment:
I think we want this & any tags that end with `-dd.*` or something like that, right? Or is the idea that we no longer would need to cut tags here and could ship the `v18411704-d9bda6da-1.12.11` images directly?

Reply:
Currently does the job get triggered every time there's a new commit ?
WDYT think about triggering CI only based on tag patterns instead ? We could use a `stg` prefix to indicate that image should be pushed only to staging.

Reply:
> is the idea that we no longer would need to cut tags here 

Yep, so my proposed idea is to change the workflow so that we stop using tags and would ship images directly from every commit on the `-dd` branches. This is just less things to do for us IMO. But I'm obviously open to feedback!

> Currently does the job get triggered every time there's a new commit ?

Not exactly:
- For all branches ending with `-dd`, automatically build and push all images to prod
- For all other branches, create manual jobs for pushing to staging (this can be used for testing)

> WDYT think about triggering CI only based on tag patterns instead ? We could use a stg prefix to indicate that image should be pushed only to staging.

This could work as well, there are many potential workflows that we could go with! My initial idea was to automate as much as possible so that we don't need to push any tags or think about tag prefixes etc. (It's something we can discuss as a team in the next sync as well)